### DISCUSSION TITLE
[3x] sprotty: findServersForUri

### DISCUSSION THREAD

**Discussion by cdietrich** [eclipse-langium/langium#1318] [Q&A]
Hi, is there any information how the new
`findServersForUri` method is meant to be used?
is the assumption there is no reuse of uris between different clients?
is there an example to get inspiration

**Comment by spoenemann** [comment]
The method gives you all DiagramServers that match a given URI according to what the respective Sprotty clients have requested. What else do you need to know?

**Reply by cdietrich** [reply]
my question is: how do i need to implement it.
given `OurDiagramServerManager implements DiagramServerManager {`

right now we have

```
// Client ID to diagram server instance
  protected readonly diagramServerMap: Map<string, DiagramServer> = new Map()
  ```
  as storage for servers
  
  so i can either return all. or need to filter

**Reply by spoenemann** [reply]
Why do you want to implement your own DiagramServerManager? Can't you extend the default impl?

**Reply by cdietrich** [reply]
need to find out.commit message just says

> This implementation aims to provide simultaneous sync between multiple views, only refreshing the diagrams once updated, while also keeping the zoom consistent. This is done by rebuilding diagram server manager.
